Error in using data source aws_ec2_managed_prefix_list and filter from a list of many prefix list ids in Managed prefix lists

Hi there,

we are not able to use data source for **aws_ec2_managed_prefix_list**. please note that we have around 600 prefix list id in Managed prefix lists and tried to filter **com.amazonaws.global.cloudfront.origin-facing** using the prefix-list-name and got the below error.
would like to know if there are any limitations 
```
Error: no managed prefix lists matched the given criteria

  with data.aws_ec2_managed_prefix_list.cloudfront,
  on data.tf line 32, in data "aws_ec2_managed_prefix_list" "cloudfront":
  32: data "aws_ec2_managed_prefix_list" "cloudfront" {
```

Upon checking the cloudtrail logs, we found **"responseElements": null,**  for one of cal we made.

```
"requestParameters": {
        "DescribeManagedPrefixListsRequest": {
            "Filter": [
                {
                    "Value": {
                        "tag": 1,
                        "content": "[com.amazonaws.global.cloudfront.origin-facing"
                    },
                    "tag": 1,
                    "Name": "prefix-list-name"
                },
                {
                    "Value": {
                        "tag": 1,
                        "content": "AWS"
                    },
                    "tag": 2,
                    "Name": "owner-id"
                }
            ]
        }
    },
    "responseElements": null,
```
<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version

**Terraform v1.2.4**
**provider registry.terraform.io/hashicorp/aws v4.22.0**

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_ec2_managed_prefix_list

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "aws_ec2_managed_prefix_list" "cloudfront" {
  filter {
    name   = "owner-id"
    values = ["AWS"]
  }

  filter {
    name   = "prefix-list-name"
    values = ["com.amazonaws.global.cloudfront.origin-facing"]
  }
}

```

### Debug Output
```
➜  ec2_prefix terraform init                                                                     <aws:awsvikingsnp_admin>
2022-07-27T07:35:29.310-0700 [INFO]  Terraform version: 1.2.4
2022-07-27T07:35:29.310-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.0.0
2022-07-27T07:35:29.310-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.12.0
2022-07-27T07:35:29.310-0700 [DEBUG] using github.com/hashicorp/terraform-config-inspect v0.0.0-20210209133302-4fd17a0faac2
2022-07-27T07:35:29.310-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.0.0-20200729002733-f050f53b9734
2022-07-27T07:35:29.310-0700 [DEBUG] using github.com/zclconf/go-cty v1.10.0
2022-07-27T07:35:29.310-0700 [INFO]  Go runtime version: go1.18.1
2022-07-27T07:35:29.310-0700 [INFO]  CLI args: []string{"/opt/homebrew/Cellar/tfenv/2.2.3/versions/1.2.4/terraform", "init"}
2022-07-27T07:35:29.310-0700 [DEBUG] Attempting to open CLI config file: /Users/arun.sandu/.terraformrc
2022-07-27T07:35:29.310-0700 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2022-07-27T07:35:29.310-0700 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2022-07-27T07:35:29.310-0700 [DEBUG] ignoring non-existing provider search directory /Users/arun.sandu/.terraform.d/plugins
2022-07-27T07:35:29.310-0700 [DEBUG] ignoring non-existing provider search directory /Users/arun.sandu/Library/Application Support/io.terraform/plugins
2022-07-27T07:35:29.310-0700 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2022-07-27T07:35:29.311-0700 [INFO]  CLI command args: []string{"init"}

Initializing the backend...
2022-07-27T07:35:29.314-0700 [DEBUG] New state was assigned lineage "3ac82b3a-1d54-8e88-6956-21cfab200b37"
2022-07-27T07:35:29.315-0700 [DEBUG] checking for provisioner in "."
2022-07-27T07:35:29.315-0700 [DEBUG] checking for provisioner in "/opt/homebrew/Cellar/tfenv/2.2.3/versions/1.2.4"

Initializing provider plugins...
- Reusing previous version of hashicorp/aws from the dependency lock file
2022-07-27T07:35:29.315-0700 [DEBUG] Service discovery for registry.terraform.io at https://registry.terraform.io/.well-known/terraform.json
2022-07-27T07:35:29.420-0700 [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/aws/versions
2022-07-27T07:35:29.487-0700 [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/aws/4.22.0/download/darwin/amd64
2022-07-27T07:35:29.548-0700 [DEBUG] GET https://releases.hashicorp.com/terraform-provider-aws/4.22.0/terraform-provider-aws_4.22.0_SHA256SUMS
2022-07-27T07:35:29.630-0700 [DEBUG] GET https://releases.hashicorp.com/terraform-provider-aws/4.22.0/terraform-provider-aws_4.22.0_SHA256SUMS.72D7468F.sig
- Installing hashicorp/aws v4.22.0...
2022-07-27T07:35:35.071-0700 [DEBUG] Provider signed by 34365D9472D7468F HashiCorp Security (hashicorp.com/security) <security@hashicorp.com>
- Installed hashicorp/aws v4.22.0 (signed by HashiCorp)

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
➜  ec2_prefix terraform plan                                                                     <aws:awsvikingsnp_admin>
2022-07-27T07:35:43.635-0700 [INFO]  Terraform version: 1.2.4
2022-07-27T07:35:43.635-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.0.0
2022-07-27T07:35:43.635-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.12.0
2022-07-27T07:35:43.635-0700 [DEBUG] using github.com/hashicorp/terraform-config-inspect v0.0.0-20210209133302-4fd17a0faac2
2022-07-27T07:35:43.635-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.0.0-20200729002733-f050f53b9734
2022-07-27T07:35:43.635-0700 [DEBUG] using github.com/zclconf/go-cty v1.10.0
2022-07-27T07:35:43.635-0700 [INFO]  Go runtime version: go1.18.1
2022-07-27T07:35:43.635-0700 [INFO]  CLI args: []string{"/opt/homebrew/Cellar/tfenv/2.2.3/versions/1.2.4/terraform", "plan"}
2022-07-27T07:35:43.635-0700 [DEBUG] Attempting to open CLI config file: /Users/arun.sandu/.terraformrc
2022-07-27T07:35:43.635-0700 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2022-07-27T07:35:43.636-0700 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2022-07-27T07:35:43.636-0700 [DEBUG] ignoring non-existing provider search directory /Users/arun.sandu/.terraform.d/plugins
2022-07-27T07:35:43.636-0700 [DEBUG] ignoring non-existing provider search directory /Users/arun.sandu/Library/Application Support/io.terraform/plugins
2022-07-27T07:35:43.638-0700 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2022-07-27T07:35:43.639-0700 [INFO]  CLI command args: []string{"plan"}
2022-07-27T07:35:43.639-0700 [DEBUG] New state was assigned lineage "7d49c03b-5fec-e811-87a9-7cec8f391ac2"
2022-07-27T07:35:44.676-0700 [DEBUG] checking for provisioner in "."
2022-07-27T07:35:44.676-0700 [DEBUG] checking for provisioner in "/opt/homebrew/Cellar/tfenv/2.2.3/versions/1.2.4"
2022-07-27T07:35:44.677-0700 [INFO]  backend/local: starting Plan operation
2022-07-27T07:35:44.679-0700 [DEBUG] created provider logger: level=debug
2022-07-27T07:35:44.679-0700 [INFO]  provider: configuring client automatic mTLS
2022-07-27T07:35:44.697-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5]
2022-07-27T07:35:44.711-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26012
2022-07-27T07:35:44.711-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5
2022-07-27T07:36:02.207-0700 [INFO]  provider.terraform-provider-aws_v4.22.0_x5: configuring server automatic mTLS: timestamp=2022-07-27T07:36:02.206-0700
2022-07-27T07:36:02.216-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: plugin address: address=/var/folders/jt/zrt856c52_dbkpltt28llwm80000gq/T/plugin3095414796 network=unix timestamp=2022-07-27T07:36:02.216-0700
2022-07-27T07:36:02.216-0700 [DEBUG] provider: using plugin: version=5
2022-07-27T07:36:02.332-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-07-27T07:36:02.337-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26012
2022-07-27T07:36:02.337-0700 [DEBUG] provider: plugin exited
2022-07-27T07:36:02.338-0700 [DEBUG] Building and walking validate graph
2022-07-27T07:36:02.338-0700 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/aws"], implied first by data.aws_ec2_managed_prefix_list.cloudfront
2022-07-27T07:36:02.338-0700 [DEBUG] ProviderTransformer: "data.aws_ec2_managed_prefix_list.cloudfront" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2022-07-27T07:36:02.338-0700 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
2022-07-27T07:36:02.338-0700 [DEBUG] ReferenceTransformer: "data.aws_ec2_managed_prefix_list.cloudfront" references: []
2022-07-27T07:36:02.338-0700 [DEBUG] Starting graph walk: walkValidate
2022-07-27T07:36:02.339-0700 [DEBUG] created provider logger: level=debug
2022-07-27T07:36:02.339-0700 [INFO]  provider: configuring client automatic mTLS
2022-07-27T07:36:02.357-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5]
2022-07-27T07:36:02.370-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26067
2022-07-27T07:36:02.370-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5
2022-07-27T07:36:02.466-0700 [INFO]  provider.terraform-provider-aws_v4.22.0_x5: configuring server automatic mTLS: timestamp=2022-07-27T07:36:02.466-0700
2022-07-27T07:36:02.475-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: plugin address: address=/var/folders/jt/zrt856c52_dbkpltt28llwm80000gq/T/plugin1597892428 network=unix timestamp=2022-07-27T07:36:02.475-0700
2022-07-27T07:36:02.475-0700 [DEBUG] provider: using plugin: version=5
2022-07-27T07:36:02.567-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-07-27T07:36:02.572-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26067
2022-07-27T07:36:02.572-0700 [DEBUG] provider: plugin exited
2022-07-27T07:36:02.573-0700 [INFO]  backend/local: plan calling Plan
2022-07-27T07:36:02.573-0700 [DEBUG] Building and walking plan graph for NormalMode
2022-07-27T07:36:02.573-0700 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/aws"], implied first by data.aws_ec2_managed_prefix_list.cloudfront (expand)
2022-07-27T07:36:02.573-0700 [DEBUG] ProviderTransformer: "data.aws_ec2_managed_prefix_list.cloudfront (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2022-07-27T07:36:02.574-0700 [DEBUG] ReferenceTransformer: "data.aws_ec2_managed_prefix_list.cloudfront (expand)" references: []
2022-07-27T07:36:02.574-0700 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
2022-07-27T07:36:02.574-0700 [DEBUG] Starting graph walk: walkPlan
2022-07-27T07:36:02.574-0700 [DEBUG] created provider logger: level=debug
2022-07-27T07:36:02.574-0700 [INFO]  provider: configuring client automatic mTLS
2022-07-27T07:36:02.593-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5]
2022-07-27T07:36:02.607-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26068
2022-07-27T07:36:02.607-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5
2022-07-27T07:36:02.708-0700 [INFO]  provider.terraform-provider-aws_v4.22.0_x5: configuring server automatic mTLS: timestamp=2022-07-27T07:36:02.707-0700
2022-07-27T07:36:02.716-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: plugin address: address=/var/folders/jt/zrt856c52_dbkpltt28llwm80000gq/T/plugin1552206433 network=unix timestamp=2022-07-27T07:36:02.716-0700
2022-07-27T07:36:02.717-0700 [DEBUG] provider: using plugin: version=5
2022-07-27T07:36:02.880-0700 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/aws\"]" changed the config value, but that value is unused
2022-07-27T07:36:02.884-0700 [INFO]  provider.terraform-provider-aws_v4.22.0_x5: [INFO] Retrieved credentials from "EnvConfigCredentials"
2022-07-27T07:36:02.884-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] Trying to get account information via sts:GetCallerIdentity
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go-v2] Request
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: POST / HTTP/1.1
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Host: sts.us-east-1.amazonaws.com
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.4 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.16.6 os/macos lang/go/1.17.6 md/GOOS/darwin md/GOARCH/amd64 api/sts/1.16.4
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 43
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Amz-Sdk-Invocation-Id: 361cc43b-b02e-4276-96b1-dfe6a2896b25
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Amz-Sdk-Request: attempt=1; max=25
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIAZV42Y5SOVLAT7X6H/20220727/us-east-1/sts/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=c6a6eb6b708e4159fec3af96231ce509786e35750cec988152729e6ce0268eef
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: application/x-www-form-urlencoded
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Date: 20220727T143602Z
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEI///////////wEaCXVzLWVhc3QtMSJHMEUCIQDoJyX6ppB2X/MAEwhPLUXpA7nxWewXkzJnh81HWRhNRAIgFr9L0HsBtaa6Nh2RcZPHbmijw1Ang+qf9DU4y3oz17sqngII2P//////////ARACGgw2NjU1MDc2NTQ4MTMiDH0nW5S49z9TVV1ScSryAe7XhXde/bYrm+hsCXDRrHffdKg7Vm8oJ4kmRo0i2zyj0CAfZ3wc+QBqMAByj/oxa4FOWbR8De0TGgDXNgtN2vO4lY9Bh5VZ397otcbJiezkEDzB98ogqKQRAjTW0Jv8a1zZhSJISlN5D00UfGca4br1vsIUpDAQlHbH3+qSBb5IZdrc+6Kvg5TnCdSZZhNMYKFM4H8PT8et9qgEP36jGKv/SGN78OXtUKZF+4tEOhdB7Uk62nodroushV9jG4qJzWzKULoejb/CfH7AKU4vEgXAZglxK07BgDA9FOUwFPyKC66Vt4cUT5eAGnVCEb5B/jcsMLaYhZcGOp0BrAmEkKOxnIo/bCj6qJip+y1U275j+4LSB1ke1PDrf5faLgHhuWHvToI9CrMMBB2A7xWwIQuczQfDxRXhYxSzQpGkSp3pPdYk+UhGsPN4543xNQSYaeg4gPRdDfdJv+aGrqjnyv00FqC7i7bx7qVT0scpHmddNIIk2quxgw8lmz0VPOFbHtMfiQMZ0yBnJ+kNpasIpNlsnG40SSyLzg==
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Accept-Encoding: gzip
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:02.887-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Action=GetCallerIdentity&Version=2011-06-15
2022-07-27T07:36:03.432-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go-v2] Response
2022-07-27T07:36:03.432-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: HTTP/1.1 200 OK
2022-07-27T07:36:03.432-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 426
2022-07-27T07:36:03.432-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: text/xml
2022-07-27T07:36:03.433-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Date: Wed, 27 Jul 2022 14:36:02 GMT
2022-07-27T07:36:03.433-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amzn-Requestid: d2937220-8bec-43c5-bba4-755565b02df7
2022-07-27T07:36:03.433-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:03.433-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: <GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
2022-07-27T07:36:03.433-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   <GetCallerIdentityResult>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <Arn>arn:aws:sts::665507654813:assumed-role/Admin/AKSANDU</Arn>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <UserId>AROAIJDED3AQYM5J6HUI2:AKSANDU</UserId>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <Account>665507654813</Account>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   </GetCallerIdentityResult>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   <ResponseMetadata>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <RequestId>d2937220-8bec-43c5-bba4-755565b02df7</RequestId>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   </ResponseMetadata>
2022-07-27T07:36:03.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: </GetCallerIdentityResponse>
2022-07-27T07:36:03.436-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] Trying to get account information via sts:GetCallerIdentity
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go-v2] Request
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: POST / HTTP/1.1
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Host: sts.us-east-1.amazonaws.com
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.4 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.16.6 os/macos lang/go/1.17.6 md/GOOS/darwin md/GOARCH/amd64 api/sts/1.16.4
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 43
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Amz-Sdk-Invocation-Id: 847e3179-12b3-43fb-a2a6-471a78b3beee
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Amz-Sdk-Request: attempt=1; max=25
2022-07-27T07:36:03.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIAZV42Y5SOVLAT7X6H/20220727/us-east-1/sts/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=b783d2b3dca39c8ffaaf376645363f16a55dedbb3106d6fc76f1ca38ce7eb9fc
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: application/x-www-form-urlencoded
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Date: 20220727T143603Z
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEI///////////wEaCXVzLWVhc3QtMSJHMEUCIQDoJyX6ppB2X/MAEwhPLUXpA7nxWewXkzJnh81HWRhNRAIgFr9L0HsBtaa6Nh2RcZPHbmijw1Ang+qf9DU4y3oz17sqngII2P//////////ARACGgw2NjU1MDc2NTQ4MTMiDH0nW5S49z9TVV1ScSryAe7XhXde/bYrm+hsCXDRrHffdKg7Vm8oJ4kmRo0i2zyj0CAfZ3wc+QBqMAByj/oxa4FOWbR8De0TGgDXNgtN2vO4lY9Bh5VZ397otcbJiezkEDzB98ogqKQRAjTW0Jv8a1zZhSJISlN5D00UfGca4br1vsIUpDAQlHbH3+qSBb5IZdrc+6Kvg5TnCdSZZhNMYKFM4H8PT8et9qgEP36jGKv/SGN78OXtUKZF+4tEOhdB7Uk62nodroushV9jG4qJzWzKULoejb/CfH7AKU4vEgXAZglxK07BgDA9FOUwFPyKC66Vt4cUT5eAGnVCEb5B/jcsMLaYhZcGOp0BrAmEkKOxnIo/bCj6qJip+y1U275j+4LSB1ke1PDrf5faLgHhuWHvToI9CrMMBB2A7xWwIQuczQfDxRXhYxSzQpGkSp3pPdYk+UhGsPN4543xNQSYaeg4gPRdDfdJv+aGrqjnyv00FqC7i7bx7qVT0scpHmddNIIk2quxgw8lmz0VPOFbHtMfiQMZ0yBnJ+kNpasIpNlsnG40SSyLzg==
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Accept-Encoding: gzip
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:03.438-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Action=GetCallerIdentity&Version=2011-06-15
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go-v2] Response
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: HTTP/1.1 200 OK
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 426
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: text/xml
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Date: Wed, 27 Jul 2022 14:36:02 GMT
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amzn-Requestid: a469ea13-9af3-40b3-b938-989e4101681c
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: <GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
2022-07-27T07:36:03.521-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   <GetCallerIdentityResult>
2022-07-27T07:36:03.537-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <Arn>arn:aws:sts::665507654813:assumed-role/Admin/AKSANDU</Arn>
2022-07-27T07:36:03.537-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <UserId>AROAIJDED3AQYM5J6HUI2:AKSANDU</UserId>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <Account>665507654813</Account>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   </GetCallerIdentityResult>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   <ResponseMetadata>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <RequestId>a469ea13-9af3-40b3-b938-989e4101681c</RequestId>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:   </ResponseMetadata>
2022-07-27T07:36:03.538-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: </GetCallerIdentityResponse>
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Request ec2/DescribeAccountAttributes Details:
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: ---[ REQUEST POST-SIGN ]-----------------------------
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: POST / HTTP/1.1
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Host: ec2.us-east-1.amazonaws.com
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.4 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.46 (go1.17.6; darwin; amd64)
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 87
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIAZV42Y5SOVLAT7X6H/20220727/us-east-1/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=5fcbaddd3c1b24525194dae420342b81e62e7c9f3113456e0e47520d527d675d
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Date: 20220727T143603Z
2022-07-27T07:36:03.556-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEI///////////wEaCXVzLWVhc3QtMSJHMEUCIQDoJyX6ppB2X/MAEwhPLUXpA7nxWewXkzJnh81HWRhNRAIgFr9L0HsBtaa6Nh2RcZPHbmijw1Ang+qf9DU4y3oz17sqngII2P//////////ARACGgw2NjU1MDc2NTQ4MTMiDH0nW5S49z9TVV1ScSryAe7XhXde/bYrm+hsCXDRrHffdKg7Vm8oJ4kmRo0i2zyj0CAfZ3wc+QBqMAByj/oxa4FOWbR8De0TGgDXNgtN2vO4lY9Bh5VZ397otcbJiezkEDzB98ogqKQRAjTW0Jv8a1zZhSJISlN5D00UfGca4br1vsIUpDAQlHbH3+qSBb5IZdrc+6Kvg5TnCdSZZhNMYKFM4H8PT8et9qgEP36jGKv/SGN78OXtUKZF+4tEOhdB7Uk62nodroushV9jG4qJzWzKULoejb/CfH7AKU4vEgXAZglxK07BgDA9FOUwFPyKC66Vt4cUT5eAGnVCEb5B/jcsMLaYhZcGOp0BrAmEkKOxnIo/bCj6qJip+y1U275j+4LSB1ke1PDrf5faLgHhuWHvToI9CrMMBB2A7xWwIQuczQfDxRXhYxSzQpGkSp3pPdYk+UhGsPN4543xNQSYaeg4gPRdDfdJv+aGrqjnyv00FqC7i7bx7qVT0scpHmddNIIk2quxgw8lmz0VPOFbHtMfiQMZ0yBnJ+kNpasIpNlsnG40SSyLzg==
2022-07-27T07:36:03.557-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Accept-Encoding: gzip
2022-07-27T07:36:03.557-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:03.557-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Action=DescribeAccountAttributes&AttributeName.1=supported-platforms&Version=2016-11-15
2022-07-27T07:36:03.557-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: -----------------------------------------------------
2022-07-27T07:36:04.114-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Response ec2/DescribeAccountAttributes Details:
2022-07-27T07:36:04.116-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: ---[ RESPONSE ]--------------------------------------
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: HTTP/1.1 200 OK
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 540
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Cache-Control: no-cache, no-store
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: text/xml;charset=UTF-8
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Date: Wed, 27 Jul 2022 14:36:03 GMT
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Server: AmazonEC2
2022-07-27T07:36:04.117-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2022-07-27T07:36:04.119-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amzn-Requestid: ae99ba37-739a-49c2-b300-72a476432bc7
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: -----------------------------------------------------
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] <?xml version="1.0" encoding="UTF-8"?>
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: <DescribeAccountAttributesResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <requestId>ae99ba37-739a-49c2-b300-72a476432bc7</requestId>
2022-07-27T07:36:04.120-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <accountAttributeSet>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:         <item>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:             <attributeName>supported-platforms</attributeName>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:             <attributeValueSet>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:                 <item>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:                     <attributeValue>VPC</attributeValue>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:                 </item>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:             </attributeValueSet>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:         </item>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     </accountAttributeSet>
2022-07-27T07:36:04.121-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: </DescribeAccountAttributesResponse>
2022-07-27T07:36:04.132-0700 [DEBUG] Resource instance state not found for node "data.aws_ec2_managed_prefix_list.cloudfront", instance data.aws_ec2_managed_prefix_list.cloudfront
2022-07-27T07:36:04.135-0700 [DEBUG] ReferenceTransformer: "data.aws_ec2_managed_prefix_list.cloudfront" references: []
data.aws_ec2_managed_prefix_list.cloudfront: Reading...
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Request ec2/DescribeManagedPrefixLists Details:
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: ---[ REQUEST POST-SIGN ]-----------------------------
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: POST / HTTP/1.1
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Host: ec2.us-east-1.amazonaws.com
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.2.4 (+https://www.terraform.io) terraform-provider-aws/dev (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.46 (go1.17.6; darwin; amd64)
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Length: 190
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Authorization: AWS4-HMAC-SHA256 Credential=ASIAZV42Y5SOVLAT7X6H/20220727/us-east-1/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=2c09eb60a1e0524183d7d05a8451ba8ce827b3dae2e3bbf0e2d970d139a1f2ad
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Date: 20220727T143604Z
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amz-Security-Token: IQoJb3JpZ2luX2VjEI///////////wEaCXVzLWVhc3QtMSJHMEUCIQDoJyX6ppB2X/MAEwhPLUXpA7nxWewXkzJnh81HWRhNRAIgFr9L0HsBtaa6Nh2RcZPHbmijw1Ang+qf9DU4y3oz17sqngII2P//////////ARACGgw2NjU1MDc2NTQ4MTMiDH0nW5S49z9TVV1ScSryAe7XhXde/bYrm+hsCXDRrHffdKg7Vm8oJ4kmRo0i2zyj0CAfZ3wc+QBqMAByj/oxa4FOWbR8De0TGgDXNgtN2vO4lY9Bh5VZ397otcbJiezkEDzB98ogqKQRAjTW0Jv8a1zZhSJISlN5D00UfGca4br1vsIUpDAQlHbH3+qSBb5IZdrc+6Kvg5TnCdSZZhNMYKFM4H8PT8et9qgEP36jGKv/SGN78OXtUKZF+4tEOhdB7Uk62nodroushV9jG4qJzWzKULoejb/CfH7AKU4vEgXAZglxK07BgDA9FOUwFPyKC66Vt4cUT5eAGnVCEb5B/jcsMLaYhZcGOp0BrAmEkKOxnIo/bCj6qJip+y1U275j+4LSB1ke1PDrf5faLgHhuWHvToI9CrMMBB2A7xWwIQuczQfDxRXhYxSzQpGkSp3pPdYk+UhGsPN4543xNQSYaeg4gPRdDfdJv+aGrqjnyv00FqC7i7bx7qVT0scpHmddNIIk2quxgw8lmz0VPOFbHtMfiQMZ0yBnJ+kNpasIpNlsnG40SSyLzg==
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Accept-Encoding: gzip
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Action=DescribeManagedPrefixLists&Filter.1.Name=prefix-list-name&Filter.1.Value.1=com.amazonaws.global.cloudfront.origin-facing&Filter.2.Name=owner-id&Filter.2.Value.1=AWS&Version=2016-11-15
2022-07-27T07:36:04.148-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: -----------------------------------------------------
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] DEBUG: Response ec2/DescribeManagedPrefixLists Details:
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: ---[ RESPONSE ]--------------------------------------
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: HTTP/1.1 200 OK
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Transfer-Encoding: chunked
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Cache-Control: no-cache, no-store
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Content-Type: text/xml;charset=UTF-8
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Date: Wed, 27 Jul 2022 14:36:04 GMT
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Server: AmazonEC2
2022-07-27T07:36:04.435-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: Vary: accept-encoding
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: X-Amzn-Requestid: 614274eb-4b98-46f2-9ff1-8cd879d0fe48
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: -----------------------------------------------------
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: [DEBUG] [aws-sdk-go] <?xml version="1.0" encoding="UTF-8"?>
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: <DescribeManagedPrefixListsResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
2022-07-27T07:36:04.437-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <requestId>614274eb-4b98-46f2-9ff1-8cd879d0fe48</requestId>
2022-07-27T07:36:04.439-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <nextToken>AYABFDzI/ZZPy+V3KN8/ZbRA8J8AAAABAAdhd3Mta21zAEthcm46YXdzOmttczp1cy1lYXN0LTE6NTU5Mjg2NTc2OTY3OmtleS8xY2MyMzEzMS1iYTIxLTQ3OTYtOWQ5Ny1jNGI4NWM4NDQ4OGUAqAECAQB4Y00ENZKPjkk6JVeIWVGcfgR3BW79lXD41cOkutJlC0gBtT1j9p6DAjoDK+WldUrQbAAAAG4wbAYJKoZIhvcNAQcGoF8wXQIBADBYBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDFqX0NH2TS4PpYshhgIBEIArRYTjKFdFpBmqTkZLSv4S6+EbWxrxrefMg4mbCw9a5Z1hF1y6SeAS5jIcuwIAAAAADAAAEAAAAAAAAAAAAAAAAABHkrhtWeDN0VJECRVFTc9g/////wAAAAEAAAAAAAAAAAAAAAEAAAIlDxsbUhEq7BGLPX6zVb/gDAFOs5bZoPs6rL3+E5JuFtwZka3b4sWvlV6NuqL3r5Y7dxX3PX44bG0l68+jKXrPDCQzmz4wGDMRmV3kJSLH4coyAs3xKRoGveJKnI5wBnytH4YpN12H26eFMcpMWNMdZwsSgS7rAqiOLbHvZ5FRDm/aHxUjcAZ28vsDqCBOPenn0rvvQF/KgHSvuRjZCafNLZS9jK3uiUFFh0i1jtqGDRY7aWeK8XX1W+N+NcPJ8YuAAGWgd8/K5wq5+5SL+GNyb20FlrxWQ9t+Xn86TsR7db4qquylsbsbPnMy/5eOsPyWKqIMO7KU9TWxAG5D4Si0Z2BtTVame9VMIQ/Nl9GoSmjcezehhmFeA0Ewiw5tdX9zSf8ShXYUp5rs7BoxDlbjDwT+VYhIk5+p2ewuKw3zBlsb0ic8uqBxudFFzrkexOdc0eR3ZbcYV0QRkDzma4ooOwartHfCJ3oOH35K6lor4KGLnZtfmfsC03yPNIZYLsi8fcpbS+K2b24soAmqsx+c2o2qkGXSTdFoRumHq0i0/Uj96ZJQyUo6S7M6ZhlIyTkqMg+7/SUdtHTdNrUSqcc1hwNet9bOIAeXBvbQGfAqoaoJVW/fjqVFz/XxZfHMwH9ahnv82sx3C+4FyVH+y+m/e5eGUPpn/Ye2OczjFwfEkf1mPQ4daP2vbnn+SMbRlJo6Al+5jjOmBIupIcG1KgB6vepzfuhBAW806ZZTRTPiWc9fklOIUQ==</nextToken>
2022-07-27T07:36:04.439-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5:     <prefixListSet/>
2022-07-27T07:36:04.439-0700 [DEBUG] provider.terraform-provider-aws_v4.22.0_x5: </DescribeManagedPrefixListsResponse>
2022-07-27T07:36:04.443-0700 [ERROR] vertex "data.aws_ec2_managed_prefix_list.cloudfront" error: no managed prefix lists matched the given criteria
2022-07-27T07:36:04.443-0700 [ERROR] vertex "data.aws_ec2_managed_prefix_list.cloudfront" error: no managed prefix lists matched the given criteria
2022-07-27T07:36:04.443-0700 [ERROR] vertex "data.aws_ec2_managed_prefix_list.cloudfront (expand)" error: no managed prefix lists matched the given criteria
2022-07-27T07:36:04.445-0700 [INFO]  backend/local: plan operation completed
╷
│ Error: no managed prefix lists matched the given criteria
│
│   with data.aws_ec2_managed_prefix_list.cloudfront,
│   on data.tf line 1, in data "aws_ec2_managed_prefix_list" "cloudfront":
│    1: data "aws_ec2_managed_prefix_list" "cloudfront" {
│
╵
2022-07-27T07:36:04.456-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-07-27T07:36:04.464-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/4.22.0/darwin_amd64/terraform-provider-aws_v4.22.0_x5 pid=26068
2022-07-27T07:36:04.464-0700 [DEBUG] provider: plugin exited
```
<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
[ec2_managed_list_api_debuglog.txt](https://github.com/hashicorp/terraform-provider-aws/files/9200212/ec2_managed_list_api_debuglog.txt)

### Expected Behavior
use data source for **aws_ec2_managed_prefix_list** and get the response

### Actual Behavior

```
Error: no managed prefix lists matched the given criteria

  with data.aws_ec2_managed_prefix_list.cloudfront,
  on data.tf line 32, in data "aws_ec2_managed_prefix_list" "cloudfront":
  32: data "aws_ec2_managed_prefix_list" "cloudfront" {
```
### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->
1. Have enough prefix list id in Managed prefix lists. we came across this issue when having 600+ prefix lists.
2. use the below 
data.tf
```
data "aws_ec2_managed_prefix_list" "cloudfront" {
  filter {
    name   = "owner-id"
    values = ["AWS"]
  }

  filter {
    name   = "prefix-list-name"
    values = ["com.amazonaws.global.cloudfront.origin-facing"]
  }
}
```
3. `terraform init`
4. `terraform plan`


### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/.changelog/26683.txt
+++ b/.changelog/26683.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+data-source/aws_ec2_managed_prefix_list: Fixes bug where an error is returned for regions with more than 100 managed prefix lists
+```

--- a/internal/service/ec2/consts.go
+++ b/internal/service/ec2/consts.go
@@ -86,6 +86,18 @@ const (
 )
 
 const (
+	managedPrefixListAddressFamilyIPv4 = "IPv4"
+	managedPrefixListAddressFamilyIPv6 = "IPv6"
+)
+
+func managedPrefixListAddressFamily_Values() []string {
+	return []string{
+		managedPrefixListAddressFamilyIPv4,
+		managedPrefixListAddressFamilyIPv6,
+	}
+}
+
+const (
 	vpnTunnelOptionsDPDTimeoutActionClear   = "clear"
 	vpnTunnelOptionsDPDTimeoutActionNone    = "none"
 	vpnTunnelOptionsDPDTimeoutActionRestart = "restart"

--- a/internal/service/ec2/errors.go
+++ b/internal/service/ec2/errors.go
@@ -91,6 +91,7 @@ const (
 	errCodeInvalidVPNGatewayAttachmentNotFound            = "InvalidVpnGatewayAttachment.NotFound"
 	errCodeInvalidVPNGatewayIDNotFound                    = "InvalidVpnGatewayID.NotFound"
 	errCodeNatGatewayNotFound                             = "NatGatewayNotFound"
+	errCodePrefixListVersionMismatch                      = "PrefixListVersionMismatch"
 	errCodeResourceNotReady                               = "ResourceNotReady"
 	errCodeSnapshotCreationPerVolumeRateExceeded          = "SnapshotCreationPerVolumeRateExceeded"
 	errCodeUnsupportedOperation                           = "UnsupportedOperation"

--- a/internal/service/ec2/errors.go
+++ b/internal/service/ec2/errors.go
@@ -64,7 +64,7 @@ const (
 	errCodeInvalidRouteTableIDNotFound                    = "InvalidRouteTableID.NotFound"
 	errCodeInvalidRouteTableIdNotFound                    = "InvalidRouteTableId.NotFound"
 	errCodeInvalidSecurityGroupIDNotFound                 = "InvalidSecurityGroupID.NotFound"
-	errCodeInvalidSecurityGroupRuleIDNotFound             = "InvalidSecurityGroupRuleId.NotFound"
+	errCodeInvalidSecurityGroupRuleIdNotFound             = "InvalidSecurityGroupRuleId.NotFound"
 	errCodeInvalidServiceName                             = "InvalidServiceName"
 	errCodeInvalidSnapshotInUse                           = "InvalidSnapshot.InUse"
 	errCodeInvalidSnapshotNotFound                        = "InvalidSnapshot.NotFound"

--- a/internal/service/ec2/find.go
+++ b/internal/service/ec2/find.go
@@ -4980,8 +4980,8 @@ func FindLaunchTemplateVersionByTwoPartKey(conn *ec2.EC2, launchTemplateID, vers
 	return output, nil
 }
 
-func FindManagedPrefixList(conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) (*ec2.ManagedPrefixList, error) {
-	output, err := FindManagedPrefixLists(conn, input)
+func FindManagedPrefixList(ctx context.Context, conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) (*ec2.ManagedPrefixList, error) {
+	output, err := FindManagedPrefixLists(ctx, conn, input)
 
 	if err != nil {
 		return nil, err
@@ -4998,10 +4998,10 @@ func FindManagedPrefixList(conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsI
 	return output[0], nil
 }
 
-func FindManagedPrefixLists(conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) ([]*ec2.ManagedPrefixList, error) {
+func FindManagedPrefixLists(ctx context.Context, conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) ([]*ec2.ManagedPrefixList, error) {
 	var output []*ec2.ManagedPrefixList
 
-	err := conn.DescribeManagedPrefixListsPages(input, func(page *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
+	err := conn.DescribeManagedPrefixListsPagesWithContext(ctx, input, func(page *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
 		if page == nil {
 			return !lastPage
 		}
@@ -5029,12 +5029,12 @@ func FindManagedPrefixLists(conn *ec2.EC2, input *ec2.DescribeManagedPrefixLists
 	return output, nil
 }
 
-func FindManagedPrefixListByID(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
+func FindManagedPrefixListByID(ctx context.Context, conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
 	input := &ec2.DescribeManagedPrefixListsInput{
 		PrefixListIds: aws.StringSlice([]string{id}),
 	}
 
-	output, err := FindManagedPrefixList(conn, input)
+	output, err := FindManagedPrefixList(ctx, conn, input)
 
 	if err != nil {
 		return nil, err

--- a/internal/service/ec2/find.go
+++ b/internal/service/ec2/find.go
@@ -5057,10 +5057,10 @@ func FindManagedPrefixListByID(ctx context.Context, conn *ec2.EC2, id string) (*
 	return output, nil
 }
 
-func FindManagedPrefixListEntries(conn *ec2.EC2, input *ec2.GetManagedPrefixListEntriesInput) ([]*ec2.PrefixListEntry, error) {
+func FindManagedPrefixListEntries(ctx context.Context, conn *ec2.EC2, input *ec2.GetManagedPrefixListEntriesInput) ([]*ec2.PrefixListEntry, error) {
 	var output []*ec2.PrefixListEntry
 
-	err := conn.GetManagedPrefixListEntriesPages(input, func(page *ec2.GetManagedPrefixListEntriesOutput, lastPage bool) bool {
+	err := conn.GetManagedPrefixListEntriesPagesWithContext(ctx, input, func(page *ec2.GetManagedPrefixListEntriesOutput, lastPage bool) bool {
 		if page == nil {
 			return !lastPage
 		}
@@ -5088,16 +5088,16 @@ func FindManagedPrefixListEntries(conn *ec2.EC2, input *ec2.GetManagedPrefixList
 	return output, nil
 }
 
-func FindManagedPrefixListEntriesByID(conn *ec2.EC2, id string) ([]*ec2.PrefixListEntry, error) {
+func FindManagedPrefixListEntriesByID(ctx context.Context, conn *ec2.EC2, id string) ([]*ec2.PrefixListEntry, error) {
 	input := &ec2.GetManagedPrefixListEntriesInput{
 		PrefixListId: aws.String(id),
 	}
 
-	return FindManagedPrefixListEntries(conn, input)
+	return FindManagedPrefixListEntries(ctx, conn, input)
 }
 
-func FindManagedPrefixListEntryByIDAndCIDR(conn *ec2.EC2, id, cidr string) (*ec2.PrefixListEntry, error) {
-	prefixListEntries, err := FindManagedPrefixListEntriesByID(conn, id)
+func FindManagedPrefixListEntryByIDAndCIDR(ctx context.Context, conn *ec2.EC2, id, cidr string) (*ec2.PrefixListEntry, error) {
+	prefixListEntries, err := FindManagedPrefixListEntriesByID(ctx, conn, id)
 
 	if err != nil {
 		return nil, err

--- a/internal/service/ec2/find.go
+++ b/internal/service/ec2/find.go
@@ -4980,62 +4980,36 @@ func FindLaunchTemplateVersionByTwoPartKey(conn *ec2.EC2, launchTemplateID, vers
 	return output, nil
 }
 
-func FindManagedPrefixListByID(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
-	input := &ec2.DescribeManagedPrefixListsInput{
-		PrefixListIds: aws.StringSlice([]string{id}),
-	}
-
-	output, err := conn.DescribeManagedPrefixLists(input)
-
-	if tfawserr.ErrCodeEquals(err, errCodeInvalidPrefixListIDNotFound) {
-		return nil, &resource.NotFoundError{
-			LastError:   err,
-			LastRequest: input,
-		}
-	}
+func FindManagedPrefixList(conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) (*ec2.ManagedPrefixList, error) {
+	output, err := FindManagedPrefixLists(conn, input)
 
 	if err != nil {
 		return nil, err
 	}
 
-	if output == nil || len(output.PrefixLists) == 0 || output.PrefixLists[0] == nil {
+	if len(output) == 0 || output[0] == nil {
 		return nil, tfresource.NewEmptyResultError(input)
 	}
 
-	if count := len(output.PrefixLists); count > 1 {
+	if count := len(output); count > 1 {
 		return nil, tfresource.NewTooManyResultsError(count, input)
 	}
 
-	prefixList := output.PrefixLists[0]
-
-	if state := aws.StringValue(prefixList.State); state == ec2.PrefixListStateDeleteComplete {
-		return nil, &resource.NotFoundError{
-			Message:     state,
-			LastRequest: input,
-		}
-	}
-
-	return prefixList, nil
+	return output[0], nil
 }
 
-func FindManagedPrefixListEntriesByID(conn *ec2.EC2, id string) ([]*ec2.PrefixListEntry, error) {
-	input := &ec2.GetManagedPrefixListEntriesInput{
-		PrefixListId: aws.String(id),
-	}
+func FindManagedPrefixLists(conn *ec2.EC2, input *ec2.DescribeManagedPrefixListsInput) ([]*ec2.ManagedPrefixList, error) {
+	var output []*ec2.ManagedPrefixList
 
-	var prefixListEntries []*ec2.PrefixListEntry
-
-	err := conn.GetManagedPrefixListEntriesPages(input, func(page *ec2.GetManagedPrefixListEntriesOutput, lastPage bool) bool {
+	err := conn.DescribeManagedPrefixListsPages(input, func(page *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
 		if page == nil {
 			return !lastPage
 		}
 
-		for _, entry := range page.Entries {
-			if entry == nil {
-				continue
+		for _, v := range page.PrefixLists {
+			if v != nil {
+				output = append(output, v)
 			}
-
-			prefixListEntries = append(prefixListEntries, entry)
 		}
 
 		return !lastPage
@@ -5052,7 +5026,74 @@ func FindManagedPrefixListEntriesByID(conn *ec2.EC2, id string) ([]*ec2.PrefixLi
 		return nil, err
 	}
 
-	return prefixListEntries, nil
+	return output, nil
+}
+
+func FindManagedPrefixListByID(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
+	input := &ec2.DescribeManagedPrefixListsInput{
+		PrefixListIds: aws.StringSlice([]string{id}),
+	}
+
+	output, err := FindManagedPrefixList(conn, input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if state := aws.StringValue(output.State); state == ec2.PrefixListStateDeleteComplete {
+		return nil, &resource.NotFoundError{
+			Message:     state,
+			LastRequest: input,
+		}
+	}
+
+	// Eventual consistency check.
+	if aws.StringValue(output.PrefixListId) != id {
+		return nil, &resource.NotFoundError{
+			LastRequest: input,
+		}
+	}
+
+	return output, nil
+}
+
+func FindManagedPrefixListEntries(conn *ec2.EC2, input *ec2.GetManagedPrefixListEntriesInput) ([]*ec2.PrefixListEntry, error) {
+	var output []*ec2.PrefixListEntry
+
+	err := conn.GetManagedPrefixListEntriesPages(input, func(page *ec2.GetManagedPrefixListEntriesOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		for _, v := range page.Entries {
+			if v != nil {
+				output = append(output, v)
+			}
+		}
+
+		return !lastPage
+	})
+
+	if tfawserr.ErrCodeEquals(err, errCodeInvalidPrefixListIDNotFound) {
+		return nil, &resource.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func FindManagedPrefixListEntriesByID(conn *ec2.EC2, id string) ([]*ec2.PrefixListEntry, error) {
+	input := &ec2.GetManagedPrefixListEntriesInput{
+		PrefixListId: aws.String(id),
+	}
+
+	return FindManagedPrefixListEntries(conn, input)
 }
 
 func FindManagedPrefixListEntryByIDAndCIDR(conn *ec2.EC2, id, cidr string) (*ec2.PrefixListEntry, error) {
@@ -5062,9 +5103,9 @@ func FindManagedPrefixListEntryByIDAndCIDR(conn *ec2.EC2, id, cidr string) (*ec2
 		return nil, err
 	}
 
-	for _, entry := range prefixListEntries {
-		if aws.StringValue(entry.Cidr) == cidr {
-			return entry, nil
+	for _, v := range prefixListEntries {
+		if aws.StringValue(v.Cidr) == cidr {
+			return v, nil
 		}
 	}
 

--- a/internal/service/ec2/id.go
+++ b/internal/service/ec2/id.go
@@ -7,24 +7,6 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/create"
 )
 
-const managedPrefixListEntryIDSeparator = ","
-
-func ManagedPrefixListEntryCreateID(prefixListID, cidrBlock string) string {
-	parts := []string{prefixListID, cidrBlock}
-	id := strings.Join(parts, managedPrefixListEntryIDSeparator)
-	return id
-}
-
-func ManagedPrefixListEntryParseID(id string) (string, string, error) {
-	parts := strings.Split(id, managedPrefixListEntryIDSeparator)
-	if len(parts) == 2 && parts[0] != "" && parts[1] != "" {
-		return parts[0], parts[1], nil
-	}
-
-	return "", "",
-		fmt.Errorf("unexpected format for ID (%q), expected prefix-list-id"+managedPrefixListEntryIDSeparator+"cidr-block", id)
-}
-
 // RouteCreateID returns a route resource ID.
 func RouteCreateID(routeTableID, destination string) string {
 	return fmt.Sprintf("r-%s%d", routeTableID, create.StringHashcode(destination))

--- a/internal/service/ec2/status.go
+++ b/internal/service/ec2/status.go
@@ -1101,9 +1101,9 @@ func StatusInternetGatewayAttachmentState(conn *ec2.EC2, internetGatewayID, vpcI
 	}
 }
 
-func StatusManagedPrefixListState(conn *ec2.EC2, id string) resource.StateRefreshFunc {
+func StatusManagedPrefixListState(ctx context.Context, conn *ec2.EC2, id string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		output, err := FindManagedPrefixListByID(conn, id)
+		output, err := FindManagedPrefixListByID(ctx, conn, id)
 
 		if tfresource.NotFound(err) {
 			return nil, "", nil

--- a/internal/service/ec2/vpc_managed_prefix_list.go
+++ b/internal/service/ec2/vpc_managed_prefix_list.go
@@ -2,12 +2,12 @@ package ec2
 
 import (
 	"context"
-	"fmt"
 	"log"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
@@ -19,10 +19,10 @@ import (
 
 func ResourceManagedPrefixList() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceManagedPrefixListCreate,
-		Read:   resourceManagedPrefixListRead,
-		Update: resourceManagedPrefixListUpdate,
-		Delete: resourceManagedPrefixListDelete,
+		CreateWithoutTimeout: resourceManagedPrefixListCreate,
+		ReadWithoutTimeout:   resourceManagedPrefixListRead,
+		UpdateWithoutTimeout: resourceManagedPrefixListUpdate,
+		DeleteWithoutTimeout: resourceManagedPrefixListDelete,
 
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
@@ -91,7 +91,7 @@ func ResourceManagedPrefixList() *schema.Resource {
 	}
 }
 
-func resourceManagedPrefixListCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
 	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
@@ -119,27 +119,27 @@ func resourceManagedPrefixListCreate(d *schema.ResourceData, meta interface{}) e
 	}
 
 	log.Printf("[DEBUG] Creating EC2 Managed Prefix List: %s", input)
-	output, err := conn.CreateManagedPrefixList(input)
+	output, err := conn.CreateManagedPrefixListWithContext(ctx, input)
 
 	if err != nil {
-		return fmt.Errorf("error creating EC2 Managed Prefix List: %w", err)
+		return diag.Errorf("creating EC2 Managed Prefix List: %s", err)
 	}
 
 	d.SetId(aws.StringValue(output.PrefixList.PrefixListId))
 
-	if _, err := WaitManagedPrefixListCreated(context.TODO(), conn, d.Id()); err != nil {
-		return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) create: %w", d.Id(), err)
+	if _, err := WaitManagedPrefixListCreated(ctx, conn, d.Id()); err != nil {
+		return diag.Errorf("waiting for EC2 Managed Prefix List (%s) create: %s", d.Id(), err)
 	}
 
-	return resourceManagedPrefixListRead(d, meta)
+	return resourceManagedPrefixListRead(ctx, d, meta)
 }
 
-func resourceManagedPrefixListRead(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
 	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
-	pl, err := FindManagedPrefixListByID(context.TODO(), conn, d.Id())
+	pl, err := FindManagedPrefixListByID(ctx, conn, d.Id())
 
 	if !d.IsNewResource() && tfresource.NotFound(err) {
 		log.Printf("[WARN] EC2 Managed Prefix List %s not found, removing from state", d.Id())
@@ -148,43 +148,40 @@ func resourceManagedPrefixListRead(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if err != nil {
-		return fmt.Errorf("error reading EC2 Managed Prefix List (%s): %w", d.Id(), err)
+		return diag.Errorf("reading EC2 Managed Prefix List (%s): %s", d.Id(), err)
 	}
 
-	prefixListEntries, err := FindManagedPrefixListEntriesByID(context.TODO(), conn, d.Id())
+	prefixListEntries, err := FindManagedPrefixListEntriesByID(ctx, conn, d.Id())
 
 	if err != nil {
-		return fmt.Errorf("error reading EC2 Managed Prefix List (%s) Entries: %w", d.Id(), err)
+		return diag.Errorf("reading EC2 Managed Prefix List (%s) Entries: %s", d.Id(), err)
 	}
 
 	d.Set("address_family", pl.AddressFamily)
 	d.Set("arn", pl.PrefixListArn)
-
 	if err := d.Set("entry", flattenPrefixListEntries(prefixListEntries)); err != nil {
-		return fmt.Errorf("error setting entry: %w", err)
+		return diag.Errorf("setting entry: %s", err)
 	}
-
 	d.Set("max_entries", pl.MaxEntries)
 	d.Set("name", pl.PrefixListName)
 	d.Set("owner_id", pl.OwnerId)
+	d.Set("version", pl.Version)
 
 	tags := KeyValueTags(pl.Tags).IgnoreAWS().IgnoreConfig(ignoreTagsConfig)
 
 	//lintignore:AWSR002
 	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
-		return fmt.Errorf("error setting tags: %w", err)
+		return diag.Errorf("setting tags: %s", err)
 	}
 
 	if err := d.Set("tags_all", tags.Map()); err != nil {
-		return fmt.Errorf("error setting tags_all: %w", err)
+		return diag.Errorf("setting tags_all: %s", err)
 	}
-
-	d.Set("version", pl.Version)
 
 	return nil
 }
 
-func resourceManagedPrefixListUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
 	if d.HasChangesExcept("tags", "tags_all") {
@@ -241,24 +238,20 @@ func resourceManagedPrefixListUpdate(d *schema.ResourceData, meta interface{}) e
 			}
 
 			if len(descriptionOnlyRemovals) > 0 {
-				_, err := conn.ModifyManagedPrefixList(&ec2.ModifyManagedPrefixListInput{
+				_, err := conn.ModifyManagedPrefixListWithContext(ctx, &ec2.ModifyManagedPrefixListInput{
 					CurrentVersion: input.CurrentVersion,
 					PrefixListId:   aws.String(d.Id()),
 					RemoveEntries:  descriptionOnlyRemovals,
 				})
 
 				if err != nil {
-					return fmt.Errorf("error updating EC2 Managed Prefix List (%s): %w", d.Id(), err)
+					return diag.Errorf("updating EC2 Managed Prefix List (%s): %s", d.Id(), err)
 				}
 
-				managedPrefixList, err := WaitManagedPrefixListModified(context.TODO(), conn, d.Id())
+				managedPrefixList, err := WaitManagedPrefixListModified(ctx, conn, d.Id())
 
 				if err != nil {
-					return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: %w", d.Id(), err)
-				}
-
-				if managedPrefixList == nil {
-					return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: empty response", d.Id())
+					return diag.Errorf("waiting for EC2 Managed Prefix List (%s) update: %s", d.Id(), err)
 				}
 
 				input.CurrentVersion = managedPrefixList.Version
@@ -278,34 +271,35 @@ func resourceManagedPrefixListUpdate(d *schema.ResourceData, meta interface{}) e
 			wait = true
 		}
 
-		_, err := conn.ModifyManagedPrefixList(input)
+		_, err := conn.ModifyManagedPrefixListWithContext(ctx, input)
 
 		if err != nil {
-			return fmt.Errorf("error updating EC2 Managed Prefix List (%s): %w", d.Id(), err)
+			return diag.Errorf("updating EC2 Managed Prefix List (%s): %s", d.Id(), err)
 		}
 
 		if wait {
-			if _, err := WaitManagedPrefixListModified(context.TODO(), conn, d.Id()); err != nil {
-				return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: %w", d.Id(), err)
+			if _, err := WaitManagedPrefixListModified(ctx, conn, d.Id()); err != nil {
+				return diag.Errorf("waiting for EC2 Managed Prefix List (%s) update: %s", d.Id(), err)
 			}
 		}
 	}
 
 	if d.HasChange("tags_all") {
 		o, n := d.GetChange("tags_all")
-		if err := UpdateTags(conn, d.Id(), o, n); err != nil {
-			return fmt.Errorf("error updating EC2 Managed Prefix List (%s) tags: %w", d.Id(), err)
+
+		if err := UpdateTagsWithContext(ctx, conn, d.Id(), o, n); err != nil {
+			return diag.Errorf("updating EC2 Managed Prefix List (%s) tags: %s", d.Id(), err)
 		}
 	}
 
-	return resourceManagedPrefixListRead(d, meta)
+	return resourceManagedPrefixListRead(ctx, d, meta)
 }
 
-func resourceManagedPrefixListDelete(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
 	log.Printf("[INFO] Deleting EC2 Managed Prefix List: %s", d.Id())
-	_, err := conn.DeleteManagedPrefixList(&ec2.DeleteManagedPrefixListInput{
+	_, err := conn.DeleteManagedPrefixListWithContext(ctx, &ec2.DeleteManagedPrefixListInput{
 		PrefixListId: aws.String(d.Id()),
 	})
 
@@ -314,11 +308,11 @@ func resourceManagedPrefixListDelete(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting EC2 Managed Prefix List (%s): %w", d.Id(), err)
+		return diag.Errorf("deleting EC2 Managed Prefix List (%s): %s", d.Id(), err)
 	}
 
-	if _, err := WaitManagedPrefixListDeleted(context.TODO(), conn, d.Id()); err != nil {
-		return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) delete: %w", d.Id(), err)
+	if _, err := WaitManagedPrefixListDeleted(ctx, conn, d.Id()); err != nil {
+		return diag.Errorf("waiting for EC2 Managed Prefix List (%s) delete: %s", d.Id(), err)
 	}
 
 	return nil

--- a/internal/service/ec2/vpc_managed_prefix_list.go
+++ b/internal/service/ec2/vpc_managed_prefix_list.go
@@ -127,7 +127,7 @@ func resourceManagedPrefixListCreate(d *schema.ResourceData, meta interface{}) e
 
 	d.SetId(aws.StringValue(output.PrefixList.PrefixListId))
 
-	if _, err := WaitManagedPrefixListCreated(conn, d.Id()); err != nil {
+	if _, err := WaitManagedPrefixListCreated(context.TODO(), conn, d.Id()); err != nil {
 		return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) create: %w", d.Id(), err)
 	}
 
@@ -139,7 +139,7 @@ func resourceManagedPrefixListRead(d *schema.ResourceData, meta interface{}) err
 	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
 	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
-	pl, err := FindManagedPrefixListByID(conn, d.Id())
+	pl, err := FindManagedPrefixListByID(context.TODO(), conn, d.Id())
 
 	if !d.IsNewResource() && tfresource.NotFound(err) {
 		log.Printf("[WARN] EC2 Managed Prefix List %s not found, removing from state", d.Id())
@@ -251,7 +251,7 @@ func resourceManagedPrefixListUpdate(d *schema.ResourceData, meta interface{}) e
 					return fmt.Errorf("error updating EC2 Managed Prefix List (%s): %w", d.Id(), err)
 				}
 
-				managedPrefixList, err := WaitManagedPrefixListModified(conn, d.Id())
+				managedPrefixList, err := WaitManagedPrefixListModified(context.TODO(), conn, d.Id())
 
 				if err != nil {
 					return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: %w", d.Id(), err)
@@ -285,7 +285,7 @@ func resourceManagedPrefixListUpdate(d *schema.ResourceData, meta interface{}) e
 		}
 
 		if wait {
-			if _, err := WaitManagedPrefixListModified(conn, d.Id()); err != nil {
+			if _, err := WaitManagedPrefixListModified(context.TODO(), conn, d.Id()); err != nil {
 				return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: %w", d.Id(), err)
 			}
 		}
@@ -317,7 +317,7 @@ func resourceManagedPrefixListDelete(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("error deleting EC2 Managed Prefix List (%s): %w", d.Id(), err)
 	}
 
-	if _, err := WaitManagedPrefixListDeleted(conn, d.Id()); err != nil {
+	if _, err := WaitManagedPrefixListDeleted(context.TODO(), conn, d.Id()); err != nil {
 		return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) delete: %w", d.Id(), err)
 	}
 

--- a/internal/service/ec2/vpc_managed_prefix_list.go
+++ b/internal/service/ec2/vpc_managed_prefix_list.go
@@ -37,12 +37,10 @@ func ResourceManagedPrefixList() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"address_family": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
-				ValidateFunc: validation.StringInSlice(
-					[]string{"IPv4", "IPv6"},
-					false),
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(managedPrefixListAddressFamily_Values(), false),
 			},
 			"arn": {
 				Type:     schema.TypeString,

--- a/internal/service/ec2/vpc_managed_prefix_list.go
+++ b/internal/service/ec2/vpc_managed_prefix_list.go
@@ -151,7 +151,7 @@ func resourceManagedPrefixListRead(d *schema.ResourceData, meta interface{}) err
 		return fmt.Errorf("error reading EC2 Managed Prefix List (%s): %w", d.Id(), err)
 	}
 
-	prefixListEntries, err := FindManagedPrefixListEntriesByID(conn, d.Id())
+	prefixListEntries, err := FindManagedPrefixListEntriesByID(context.TODO(), conn, d.Id())
 
 	if err != nil {
 		return fmt.Errorf("error reading EC2 Managed Prefix List (%s) Entries: %w", d.Id(), err)

--- a/internal/service/ec2/vpc_managed_prefix_list_data_source.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_data_source.go
@@ -94,21 +94,32 @@ func dataSourceManagedPrefixListRead(ctx context.Context, d *schema.ResourceData
 		})
 	}
 
-	out, err := conn.DescribeManagedPrefixListsWithContext(ctx, &input)
+	var prefixLists []*ec2.ManagedPrefixList
+
+	err := conn.DescribeManagedPrefixListsPagesWithContext(
+		ctx,
+		&input,
+		func(output *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
+			for _, prefixList := range output.PrefixLists {
+				prefixLists = append(prefixLists, prefixList)
+			}
+
+			return !lastPage
+		})
 
 	if err != nil {
 		return diag.Errorf("error describing EC2 Managed Prefix Lists: %s", err)
 	}
 
-	if out == nil || len(out.PrefixLists) < 1 || out.PrefixLists[0] == nil {
+	if len(prefixLists) < 1 || prefixLists[0] == nil {
 		return diag.Errorf("no managed prefix lists matched the given criteria")
 	}
 
-	if len(out.PrefixLists) > 1 {
+	if len(prefixLists) > 1 {
 		return diag.Errorf("more than 1 prefix list matched the given criteria")
 	}
 
-	pl := out.PrefixLists[0]
+	pl := prefixLists[0]
 
 	d.SetId(aws.StringValue(pl.PrefixListId))
 	d.Set("name", pl.PrefixListName)

--- a/internal/service/ec2/vpc_managed_prefix_list_data_source.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_data_source.go
@@ -10,11 +10,12 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 )
 
 func DataSourceManagedPrefixList() *schema.Resource {
 	return &schema.Resource{
-		ReadContext: dataSourceManagedPrefixListRead,
+		ReadWithoutTimeout: dataSourceManagedPrefixListRead,
 
 		Timeouts: &schema.ResourceTimeout{
 			Read: schema.DefaultTimeout(20 * time.Minute),
@@ -45,7 +46,7 @@ func DataSourceManagedPrefixList() *schema.Resource {
 					},
 				},
 			},
-			"filter": DataSourceFiltersSchema(),
+			"filter": CustomFiltersSchema(),
 			"id": {
 				Type:     schema.TypeString,
 				Computed: true,
@@ -77,83 +78,51 @@ func dataSourceManagedPrefixListRead(ctx context.Context, d *schema.ResourceData
 	conn := meta.(*conns.AWSClient).EC2Conn
 	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
-	input := ec2.DescribeManagedPrefixListsInput{}
-
-	if filters, ok := d.GetOk("filter"); ok {
-		input.Filters = BuildFiltersDataSource(filters.(*schema.Set))
+	input := &ec2.DescribeManagedPrefixListsInput{
+		Filters: BuildAttributeFilterList(map[string]string{
+			"prefix-list-name": d.Get("name").(string),
+		}),
 	}
 
-	if prefixListId, ok := d.GetOk("id"); ok {
-		input.PrefixListIds = aws.StringSlice([]string{prefixListId.(string)})
+	if v, ok := d.GetOk("id"); ok {
+		input.PrefixListIds = aws.StringSlice([]string{v.(string)})
 	}
 
-	if prefixListName, ok := d.GetOk("name"); ok {
-		input.Filters = append(input.Filters, &ec2.Filter{
-			Name:   aws.String("prefix-list-name"),
-			Values: aws.StringSlice([]string{prefixListName.(string)}),
-		})
+	input.Filters = append(input.Filters, BuildCustomFilterList(
+		d.Get("filter").(*schema.Set),
+	)...)
+
+	if len(input.Filters) == 0 {
+		// Don't send an empty filters list; the EC2 API won't accept it.
+		input.Filters = nil
 	}
 
-	var prefixLists []*ec2.ManagedPrefixList
-
-	err := conn.DescribeManagedPrefixListsPagesWithContext(
-		ctx,
-		&input,
-		func(output *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
-			prefixLists = append(prefixLists, output.PrefixLists...)
-			return !lastPage
-		})
+	pl, err := FindManagedPrefixList(ctx, conn, input)
 
 	if err != nil {
-		return diag.Errorf("error describing EC2 Managed Prefix Lists: %s", err)
+		return diag.FromErr(tfresource.SingularDataSourceFindError("EC2 Managed Prefix List", err))
 	}
-
-	if len(prefixLists) < 1 || prefixLists[0] == nil {
-		return diag.Errorf("no managed prefix lists matched the given criteria")
-	}
-
-	if len(prefixLists) > 1 {
-		return diag.Errorf("more than 1 prefix list matched the given criteria")
-	}
-
-	pl := prefixLists[0]
 
 	d.SetId(aws.StringValue(pl.PrefixListId))
-	d.Set("name", pl.PrefixListName)
-	d.Set("owner_id", pl.OwnerId)
+
+	prefixListEntries, err := FindManagedPrefixListEntriesByID(ctx, conn, d.Id())
+
+	if err != nil {
+		return diag.Errorf("reading EC2 Managed Prefix List (%s) Entries: %s", d.Id(), err)
+	}
+
 	d.Set("address_family", pl.AddressFamily)
 	d.Set("arn", pl.PrefixListArn)
+	if err := d.Set("entries", flattenPrefixListEntries(prefixListEntries)); err != nil {
+		return diag.Errorf("setting entries: %s", err)
+	}
 	d.Set("max_entries", pl.MaxEntries)
+	d.Set("name", pl.PrefixListName)
+	d.Set("owner_id", pl.OwnerId)
 	d.Set("version", pl.Version)
 
 	if err := d.Set("tags", KeyValueTags(pl.Tags).IgnoreAWS().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
-		return diag.Errorf("error setting tags attribute: %s", err)
-	}
-
-	var entries []interface{}
-
-	err = conn.GetManagedPrefixListEntriesPages(
-		&ec2.GetManagedPrefixListEntriesInput{
-			PrefixListId: pl.PrefixListId,
-		},
-		func(output *ec2.GetManagedPrefixListEntriesOutput, lastPage bool) bool {
-			for _, entry := range output.Entries {
-				entries = append(entries, map[string]interface{}{
-					"cidr":        aws.StringValue(entry.Cidr),
-					"description": aws.StringValue(entry.Description),
-				})
-			}
-
-			return !lastPage
-		},
-	)
-
-	if err != nil {
-		return diag.Errorf("error listing EC2 Managed Prefix List (%s) entries: %s", d.Id(), err)
-	}
-
-	if err := d.Set("entries", entries); err != nil {
-		return diag.FromErr(err)
+		return diag.Errorf("setting tags: %s", err)
 	}
 
 	return nil

--- a/internal/service/ec2/vpc_managed_prefix_list_data_source.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_data_source.go
@@ -100,10 +100,7 @@ func dataSourceManagedPrefixListRead(ctx context.Context, d *schema.ResourceData
 		ctx,
 		&input,
 		func(output *ec2.DescribeManagedPrefixListsOutput, lastPage bool) bool {
-			for _, prefixList := range output.PrefixLists {
-				prefixLists = append(prefixLists, prefixList)
-			}
-
+			prefixLists = append(prefixLists, output.PrefixLists...)
 			return !lastPage
 		})
 

--- a/internal/service/ec2/vpc_managed_prefix_list_data_source_test.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_data_source_test.go
@@ -159,7 +159,7 @@ func TestAccVPCManagedPrefixListDataSource_matchesTooMany(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config:      testAccVPCManagedPrefixListDataSourceConfig_matchesTooMany,
-				ExpectError: regexp.MustCompile(`more than 1 prefix list matched the given criteria`),
+				ExpectError: regexp.MustCompile(`multiple EC2 Managed Prefix Lists matched`),
 			},
 		},
 	})

--- a/internal/service/ec2/vpc_managed_prefix_list_entry.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
@@ -52,7 +53,7 @@ func resourceManagedPrefixListEntryCreate(ctx context.Context, d *schema.Resourc
 
 	cidr := d.Get("cidr").(string)
 	plID := d.Get("prefix_list_id").(string)
-	id := ManagedPrefixListEntryCreateID(plID, cidr)
+	id := ManagedPrefixListEntryCreateResourceID(plID, cidr)
 
 	addPrefixListEntry := &ec2.AddPrefixListEntry{Cidr: aws.String(cidr)}
 
@@ -96,7 +97,7 @@ func resourceManagedPrefixListEntryCreate(ctx context.Context, d *schema.Resourc
 func resourceManagedPrefixListEntryRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
-	plID, cidr, err := ManagedPrefixListEntryParseID(d.Id())
+	plID, cidr, err := ManagedPrefixListEntryParseResourceID(d.Id())
 
 	if err != nil {
 		return diag.FromErr(err)
@@ -127,7 +128,7 @@ func resourceManagedPrefixListEntryRead(ctx context.Context, d *schema.ResourceD
 func resourceManagedPrefixListEntryDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
-	plID, cidr, err := ManagedPrefixListEntryParseID(d.Id())
+	plID, cidr, err := ManagedPrefixListEntryParseResourceID(d.Id())
 
 	if err != nil {
 		return diag.FromErr(err)
@@ -167,7 +168,7 @@ func resourceManagedPrefixListEntryDelete(ctx context.Context, d *schema.Resourc
 }
 
 func resourceManagedPrefixListEntryImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	plID, cidr, err := ManagedPrefixListEntryParseID(d.Id())
+	plID, cidr, err := ManagedPrefixListEntryParseResourceID(d.Id())
 
 	if err != nil {
 		return nil, err
@@ -177,4 +178,23 @@ func resourceManagedPrefixListEntryImport(d *schema.ResourceData, meta interface
 	d.Set("prefix_list_id", plID)
 
 	return []*schema.ResourceData{d}, nil
+}
+
+const managedPrefixListEntryIDSeparator = ","
+
+func ManagedPrefixListEntryCreateResourceID(prefixListID, cidrBlock string) string {
+	parts := []string{prefixListID, cidrBlock}
+	id := strings.Join(parts, managedPrefixListEntryIDSeparator)
+
+	return id
+}
+
+func ManagedPrefixListEntryParseResourceID(id string) (string, string, error) {
+	parts := strings.Split(id, managedPrefixListEntryIDSeparator)
+
+	if len(parts) == 2 && parts[0] != "" && parts[1] != "" {
+		return parts[0], parts[1], nil
+	}
+
+	return "", "", fmt.Errorf("unexpected format for ID (%[1]s), expected prefix-list-id%[2]scidr-block", id, managedPrefixListEntryIDSeparator)
 }

--- a/internal/service/ec2/vpc_managed_prefix_list_entry.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry.go
@@ -7,6 +7,7 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
@@ -16,9 +17,10 @@ import (
 func ResourceManagedPrefixListEntry() *schema.Resource {
 	//lintignore:R011
 	return &schema.Resource{
-		Create: resourceManagedPrefixListEntryCreate,
-		Read:   resourceManagedPrefixListEntryRead,
-		Delete: resourceManagedPrefixListEntryDelete,
+		CreateWithoutTimeout: resourceManagedPrefixListEntryCreate,
+		ReadWithoutTimeout:   resourceManagedPrefixListEntryRead,
+		DeleteWithoutTimeout: resourceManagedPrefixListEntryDelete,
+
 		Importer: &schema.ResourceImporter{
 			State: resourceManagedPrefixListEntryImport,
 		},
@@ -45,7 +47,7 @@ func ResourceManagedPrefixListEntry() *schema.Resource {
 	}
 }
 
-func resourceManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListEntryCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
 	cidr := d.Get("cidr").(string)
@@ -58,15 +60,15 @@ func resourceManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface
 		addPrefixListEntry.Description = aws.String(v.(string))
 	}
 
-	_, err := tfresource.RetryWhenAWSErrCodeEquals(d.Timeout(schema.TimeoutCreate), func() (interface{}, error) {
+	_, err := tfresource.RetryWhenAWSErrCodeEqualsContext(ctx, d.Timeout(schema.TimeoutCreate), func() (interface{}, error) {
 		mutexKey := fmt.Sprintf("vpc-managed-prefix-list-%s", plID)
 		conns.GlobalMutexKV.Lock(mutexKey)
 		defer conns.GlobalMutexKV.Unlock(mutexKey)
 
-		pl, err := FindManagedPrefixListByID(context.TODO(), conn, plID)
+		pl, err := FindManagedPrefixListByID(ctx, conn, plID)
 
 		if err != nil {
-			return nil, fmt.Errorf("error reading VPC Managed Prefix List (%s): %w", plID, err)
+			return nil, fmt.Errorf("reading VPC Managed Prefix List (%s): %w", plID, err)
 		}
 
 		input := &ec2.ModifyManagedPrefixListInput{
@@ -75,33 +77,33 @@ func resourceManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface
 			PrefixListId:   aws.String(plID),
 		}
 
-		return conn.ModifyManagedPrefixList(input)
-	}, "IncorrectState", "PrefixListVersionMismatch")
+		return conn.ModifyManagedPrefixListWithContext(ctx, input)
+	}, errCodeIncorrectState, errCodePrefixListVersionMismatch)
 
 	if err != nil {
-		return fmt.Errorf("error creating VPC Managed Prefix List Entry (%s): %w", id, err)
+		return diag.Errorf("creating VPC Managed Prefix List Entry (%s): %s", id, err)
 	}
 
 	d.SetId(id)
 
-	if _, err := WaitManagedPrefixListModified(context.TODO(), conn, plID); err != nil {
-		return fmt.Errorf("error waiting for VPC Managed Prefix List Entry (%s) create: %w", d.Id(), err)
+	if _, err := WaitManagedPrefixListModified(ctx, conn, plID); err != nil {
+		return diag.Errorf("waiting for VPC Managed Prefix List Entry (%s) create: %s", d.Id(), err)
 	}
 
-	return resourceManagedPrefixListEntryRead(d, meta)
+	return resourceManagedPrefixListEntryRead(ctx, d, meta)
 }
 
-func resourceManagedPrefixListEntryRead(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListEntryRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
 	plID, cidr, err := ManagedPrefixListEntryParseID(d.Id())
 
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	outputRaw, err := tfresource.RetryWhenNewResourceNotFound(ManagedPrefixListEntryCreateTimeout, func() (interface{}, error) {
-		return FindManagedPrefixListEntryByIDAndCIDR(context.TODO(), conn, plID, cidr)
+	outputRaw, err := tfresource.RetryWhenNewResourceNotFoundContext(ctx, ManagedPrefixListEntryCreateTimeout, func() (interface{}, error) {
+		return FindManagedPrefixListEntryByIDAndCIDR(ctx, conn, plID, cidr)
 	}, d.IsNewResource())
 
 	if !d.IsNewResource() && tfresource.NotFound(err) {
@@ -111,7 +113,7 @@ func resourceManagedPrefixListEntryRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if err != nil {
-		return fmt.Errorf("error reading VPC Managed Prefix List Entry (%s): %w", d.Id(), err)
+		return diag.Errorf("reading VPC Managed Prefix List Entry (%s): %s", d.Id(), err)
 	}
 
 	entry := outputRaw.(*ec2.PrefixListEntry)
@@ -122,24 +124,24 @@ func resourceManagedPrefixListEntryRead(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
-func resourceManagedPrefixListEntryDelete(d *schema.ResourceData, meta interface{}) error {
+func resourceManagedPrefixListEntryDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*conns.AWSClient).EC2Conn
 
 	plID, cidr, err := ManagedPrefixListEntryParseID(d.Id())
 
 	if err != nil {
-		return fmt.Errorf("error parsing VPC Managed Prefix List Entry ID (%s): %w", d.Id(), err)
+		return diag.FromErr(err)
 	}
 
-	_, err = tfresource.RetryWhenAWSErrCodeEquals(d.Timeout(schema.TimeoutCreate), func() (interface{}, error) {
+	_, err = tfresource.RetryWhenAWSErrCodeEqualsContext(ctx, d.Timeout(schema.TimeoutCreate), func() (interface{}, error) {
 		mutexKey := fmt.Sprintf("vpc-managed-prefix-list-%s", plID)
 		conns.GlobalMutexKV.Lock(mutexKey)
 		defer conns.GlobalMutexKV.Unlock(mutexKey)
 
-		pl, err := FindManagedPrefixListByID(context.TODO(), conn, plID)
+		pl, err := FindManagedPrefixListByID(ctx, conn, plID)
 
 		if err != nil {
-			return nil, fmt.Errorf("error reading VPC Managed Prefix List (%s): %w", plID, err)
+			return nil, fmt.Errorf("reading VPC Managed Prefix List (%s): %w", plID, err)
 		}
 
 		input := &ec2.ModifyManagedPrefixListInput{
@@ -148,17 +150,17 @@ func resourceManagedPrefixListEntryDelete(d *schema.ResourceData, meta interface
 			RemoveEntries:  []*ec2.RemovePrefixListEntry{{Cidr: aws.String(cidr)}},
 		}
 
-		return conn.ModifyManagedPrefixList(input)
-	}, "IncorrectState", "PrefixListVersionMismatch")
+		return conn.ModifyManagedPrefixListWithContext(ctx, input)
+	}, errCodeIncorrectState, errCodePrefixListVersionMismatch)
 
 	if err != nil {
-		return fmt.Errorf("error deleting VPC Managed Prefix List Entry (%s): %w", d.Id(), err)
+		return diag.Errorf("deleting VPC Managed Prefix List Entry (%s): %s", d.Id(), err)
 	}
 
-	_, err = WaitManagedPrefixListModified(context.TODO(), conn, plID)
+	_, err = WaitManagedPrefixListModified(ctx, conn, plID)
 
 	if err != nil {
-		return fmt.Errorf("error waiting for VPC Managed Prefix List Entry (%s) delete: %w", d.Id(), err)
+		return diag.Errorf("waiting for VPC Managed Prefix List Entry (%s) delete: %s", d.Id(), err)
 	}
 
 	return nil

--- a/internal/service/ec2/vpc_managed_prefix_list_entry.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry.go
@@ -101,7 +101,7 @@ func resourceManagedPrefixListEntryRead(d *schema.ResourceData, meta interface{}
 	}
 
 	outputRaw, err := tfresource.RetryWhenNewResourceNotFound(ManagedPrefixListEntryCreateTimeout, func() (interface{}, error) {
-		return FindManagedPrefixListEntryByIDAndCIDR(conn, plID, cidr)
+		return FindManagedPrefixListEntryByIDAndCIDR(context.TODO(), conn, plID, cidr)
 	}, d.IsNewResource())
 
 	if !d.IsNewResource() && tfresource.NotFound(err) {

--- a/internal/service/ec2/vpc_managed_prefix_list_entry.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry.go
@@ -1,6 +1,7 @@
 package ec2
 
 import (
+	"context"
 	"fmt"
 	"log"
 
@@ -62,7 +63,7 @@ func resourceManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface
 		conns.GlobalMutexKV.Lock(mutexKey)
 		defer conns.GlobalMutexKV.Unlock(mutexKey)
 
-		pl, err := FindManagedPrefixListByID(conn, plID)
+		pl, err := FindManagedPrefixListByID(context.TODO(), conn, plID)
 
 		if err != nil {
 			return nil, fmt.Errorf("error reading VPC Managed Prefix List (%s): %w", plID, err)
@@ -83,7 +84,7 @@ func resourceManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface
 
 	d.SetId(id)
 
-	if _, err := WaitManagedPrefixListModified(conn, plID); err != nil {
+	if _, err := WaitManagedPrefixListModified(context.TODO(), conn, plID); err != nil {
 		return fmt.Errorf("error waiting for VPC Managed Prefix List Entry (%s) create: %w", d.Id(), err)
 	}
 
@@ -135,7 +136,7 @@ func resourceManagedPrefixListEntryDelete(d *schema.ResourceData, meta interface
 		conns.GlobalMutexKV.Lock(mutexKey)
 		defer conns.GlobalMutexKV.Unlock(mutexKey)
 
-		pl, err := FindManagedPrefixListByID(conn, plID)
+		pl, err := FindManagedPrefixListByID(context.TODO(), conn, plID)
 
 		if err != nil {
 			return nil, fmt.Errorf("error reading VPC Managed Prefix List (%s): %w", plID, err)
@@ -154,7 +155,7 @@ func resourceManagedPrefixListEntryDelete(d *schema.ResourceData, meta interface
 		return fmt.Errorf("error deleting VPC Managed Prefix List Entry (%s): %w", d.Id(), err)
 	}
 
-	_, err = WaitManagedPrefixListModified(conn, plID)
+	_, err = WaitManagedPrefixListModified(context.TODO(), conn, plID)
 
 	if err != nil {
 		return fmt.Errorf("error waiting for VPC Managed Prefix List Entry (%s) delete: %w", d.Id(), err)

--- a/internal/service/ec2/vpc_managed_prefix_list_entry_test.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry_test.go
@@ -206,7 +206,7 @@ func testAccCheckManagedPrefixListEntryDestroy(s *terraform.State) error {
 			continue
 		}
 
-		plID, cidr, err := tfec2.ManagedPrefixListEntryParseID(rs.Primary.ID)
+		plID, cidr, err := tfec2.ManagedPrefixListEntryParseResourceID(rs.Primary.ID)
 
 		if err != nil {
 			return err
@@ -241,7 +241,7 @@ func testAccCheckManagedPrefixListEntryExists(n string, v *ec2.PrefixListEntry) 
 
 		conn := acctest.Provider.Meta().(*conns.AWSClient).EC2Conn
 
-		plID, cidr, err := tfec2.ManagedPrefixListEntryParseID(rs.Primary.ID)
+		plID, cidr, err := tfec2.ManagedPrefixListEntryParseResourceID(rs.Primary.ID)
 
 		if err != nil {
 			return err
@@ -269,7 +269,7 @@ func testAccManagedPrefixListEntryImportStateIdFunc(resourceName string) resourc
 		plID := rs.Primary.Attributes["prefix_list_id"]
 		cidr := rs.Primary.Attributes["cidr"]
 
-		return tfec2.ManagedPrefixListEntryCreateID(plID, cidr), nil
+		return tfec2.ManagedPrefixListEntryCreateResourceID(plID, cidr), nil
 	}
 }
 

--- a/internal/service/ec2/vpc_managed_prefix_list_entry_test.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_entry_test.go
@@ -1,6 +1,7 @@
 package ec2_test
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 	"testing"
@@ -211,7 +212,7 @@ func testAccCheckManagedPrefixListEntryDestroy(s *terraform.State) error {
 			return err
 		}
 
-		_, err = tfec2.FindManagedPrefixListEntryByIDAndCIDR(conn, plID, cidr)
+		_, err = tfec2.FindManagedPrefixListEntryByIDAndCIDR(context.Background(), conn, plID, cidr)
 
 		if tfresource.NotFound(err) {
 			continue
@@ -246,7 +247,7 @@ func testAccCheckManagedPrefixListEntryExists(n string, v *ec2.PrefixListEntry) 
 			return err
 		}
 
-		output, err := tfec2.FindManagedPrefixListEntryByIDAndCIDR(conn, plID, cidr)
+		output, err := tfec2.FindManagedPrefixListEntryByIDAndCIDR(context.Background(), conn, plID, cidr)
 
 		if err != nil {
 			return err

--- a/internal/service/ec2/vpc_managed_prefix_list_test.go
+++ b/internal/service/ec2/vpc_managed_prefix_list_test.go
@@ -1,6 +1,7 @@
 package ec2_test
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 	"testing"
@@ -310,7 +311,7 @@ func testAccCheckManagedPrefixListDestroy(s *terraform.State) error {
 			continue
 		}
 
-		_, err := tfec2.FindManagedPrefixListByID(conn, rs.Primary.ID)
+		_, err := tfec2.FindManagedPrefixListByID(context.Background(), conn, rs.Primary.ID)
 
 		if tfresource.NotFound(err) {
 			continue
@@ -339,7 +340,7 @@ func testAccManagedPrefixListExists(resourceName string) resource.TestCheckFunc 
 
 		conn := acctest.Provider.Meta().(*conns.AWSClient).EC2Conn
 
-		_, err := tfec2.FindManagedPrefixListByID(conn, rs.Primary.ID)
+		_, err := tfec2.FindManagedPrefixListByID(context.Background(), conn, rs.Primary.ID)
 
 		return err
 	}

--- a/internal/service/ec2/vpc_security_group.go
+++ b/internal/service/ec2/vpc_security_group.go
@@ -476,7 +476,7 @@ func forceRevokeSecurityGroupRules(conn *ec2.EC2, id string, searchAll bool) err
 			_, err = conn.RevokeSecurityGroupEgress(input)
 		}
 
-		if tfawserr.ErrCodeEquals(err, errCodeInvalidSecurityGroupRuleIDNotFound) {
+		if tfawserr.ErrCodeEquals(err, errCodeInvalidSecurityGroupRuleIdNotFound) {
 			continue
 		}
 

--- a/internal/service/ec2/wait.go
+++ b/internal/service/ec2/wait.go
@@ -2281,15 +2281,15 @@ const (
 	ManagedPrefixListTimeout = 15 * time.Minute
 )
 
-func WaitManagedPrefixListCreated(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
+func WaitManagedPrefixListCreated(ctx context.Context, conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{ec2.PrefixListStateCreateInProgress},
 		Target:  []string{ec2.PrefixListStateCreateComplete},
 		Timeout: ManagedPrefixListTimeout,
-		Refresh: StatusManagedPrefixListState(conn, id),
+		Refresh: StatusManagedPrefixListState(ctx, conn, id),
 	}
 
-	outputRaw, err := stateConf.WaitForState()
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
 
 	if output, ok := outputRaw.(*ec2.ManagedPrefixList); ok {
 		if state := aws.StringValue(output.State); state == ec2.PrefixListStateCreateFailed {
@@ -2302,15 +2302,15 @@ func WaitManagedPrefixListCreated(conn *ec2.EC2, id string) (*ec2.ManagedPrefixL
 	return nil, err
 }
 
-func WaitManagedPrefixListModified(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
+func WaitManagedPrefixListModified(ctx context.Context, conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{ec2.PrefixListStateModifyInProgress},
 		Target:  []string{ec2.PrefixListStateModifyComplete},
 		Timeout: ManagedPrefixListTimeout,
-		Refresh: StatusManagedPrefixListState(conn, id),
+		Refresh: StatusManagedPrefixListState(ctx, conn, id),
 	}
 
-	outputRaw, err := stateConf.WaitForState()
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
 
 	if output, ok := outputRaw.(*ec2.ManagedPrefixList); ok {
 		if state := aws.StringValue(output.State); state == ec2.PrefixListStateModifyFailed {
@@ -2323,15 +2323,15 @@ func WaitManagedPrefixListModified(conn *ec2.EC2, id string) (*ec2.ManagedPrefix
 	return nil, err
 }
 
-func WaitManagedPrefixListDeleted(conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
+func WaitManagedPrefixListDeleted(ctx context.Context, conn *ec2.EC2, id string) (*ec2.ManagedPrefixList, error) {
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{ec2.PrefixListStateDeleteInProgress},
 		Target:  []string{},
 		Timeout: ManagedPrefixListTimeout,
-		Refresh: StatusManagedPrefixListState(conn, id),
+		Refresh: StatusManagedPrefixListState(ctx, conn, id),
 	}
 
-	outputRaw, err := stateConf.WaitForState()
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
 
 	if output, ok := outputRaw.(*ec2.ManagedPrefixList); ok {
 		if state := aws.StringValue(output.State); state == ec2.PrefixListStateDeleteFailed {

